src/command/restore/restore.c

```diff
@@ -528,6 +528,11 @@ restoreManifestMap(Manifest *const manifest)
                     // Remap tablespace if a mapping was found
                     if (tablespacePath != NULL)
                     {
+                        // We'll need to append dbid to the path for Greenplum. Currently, nothing really stores a correct dbid of
+                        // a segment that was used, so just retrieve the last folder as a segment number.
+                        if (cfgOptionStrId(cfgOptFork) == CFGOPTVAL_FORK_GPDB)
+                            tablespacePath = strNewFmt("%s/%s", strZ(tablespacePath), strBaseZ(target->path));
```

Comment:
What about using strCatChr+strCatZ or strCatFmt to don't allocate memory for a new String structure and don't copy the tablespacePath content?

Reply:
Then we would need to remove `const` from `tablespacePath` declaration, adding an extra hunk to changes.
```
error: passing argument 1 of 'strCatFmt' discards 'const' qualifier from pointer target type
```